src/contexts/bridge.context.tsx

```diff
@@ -250,6 +260,137 @@ const BridgeProvider: FC = (props) => {
+      if (from === undefined) {
+        throw new Error(
+          `The specified network_id "${network_id}" can not be found in the list of supported Chains`
+        );
+      }
+
+      const to = env.chains.find((chain) => chain.networkId === dest_net);
+      if (to === undefined) {
+        throw new Error(
+          `The specified dest_net "${dest_net}" can not be found in the list of supported Chains`
+        );
+      }
+
+      const token = await getToken({
+        env,
+        tokenAddress: token_addr,
+        originNetwork: orig_net,
+        chain: from,
+      });
+
+      const claim: Deposit["claim"] =
+        claim_tx_hash !== null
+          ? { status: "claimed", txHash: claim_tx_hash }
+          : ready_for_claim
+          ? { status: "ready" }
+          : { status: "pending" };
+
+      const tokenPrice: BigNumber | undefined = await getTokenPrice({
+        token,
+        chain: from,
+      }).catch(() => undefined);
+
+      const fiatAmount =
+        tokenPrice &&
+        multiplyAmounts(
+          {
+            value: tokenPrice,
+            precision: FIAT_DISPLAY_PRECISION,
+          },
+          {
+            value: BigNumber.from(amount),
+            precision: token.decimals,
+          },
+          FIAT_DISPLAY_PRECISION
+        );
+
+      const id = serializeBridgeId({
+        depositCount,
+        networkId,
+      });
+
+      switch (claim.status) {
+        case "pending": {
+          return {
+            status: "initiated",
+            id,
+            from,
+            to,
+            token,
+            fiatAmount,
+            amount: BigNumber.from(amount),
+            destinationAddress: dest_addr,
+            depositCount: deposit_cnt,
+            depositTxHash: tx_hash,
+            tokenOriginNetwork: orig_net,
+          };
+        }
+        case "ready": {
+          return {
+            status: "on-hold",
+            id,
+            from,
+            to,
+            token,
+            fiatAmount,
+            amount: BigNumber.from(amount),
+            destinationAddress: dest_addr,
+            depositCount: deposit_cnt,
+            depositTxHash: tx_hash,
+            tokenOriginNetwork: orig_net,
+            merkleProof: await getMerkleProof({
+              apiUrl,
+              networkId,
+              depositCount,
+            }),
+          };
+        }
+        case "claimed": {
+          return {
+            status: "completed",
+            id,
+            from,
+            to,
+            token,
+            fiatAmount,
+            amount: BigNumber.from(amount),
+            destinationAddress: dest_addr,
+            depositCount: deposit_cnt,
+            depositTxHash: tx_hash,
+            tokenOriginNetwork: orig_net,
+            claimTxHash: claim.txHash,
+          };
+        }
+      }
+    },
+    [getTokenPrice, getToken]
+  );
+
   const getBridges = useCallback(
```

Comment:
I think that the conversion from `apiDeposit` to `Deposit` is the same as in the `getBridge` function and we can abstract this a bit more later right?